documentation/book/appendix_metrics.adoc

```diff
@@ -8,74 +8,69 @@ ifdef::InstallationAppendix[]
 When adding Prometheus and Grafana servers to an Apache Kafka deployment using `minikube` or `minishift`, the memory available to the virtual machine should be increased (to 4 GB of RAM, for example, instead of the default 2 GB). Information on how to increase the default amount of memory can be found in the following section <<installing_kubernetes_and_openshift_cluster>>.
 endif::InstallationAppendix[]
 
-=== Deploying on {OpenShiftName}
+=== Prometheus
 
-==== Prometheus
+The provided Prometheus https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/{ProductVersion}/metrics/examples/prometheus/kubernetes.yaml[`kubernetes.yaml`] YAML file describes all the resources required by Prometheus in order to effectively monitor a Strimzi Kafka & ZooKeeper cluster.
+It defines the following:
 
-The Prometheus server configuration uses a service discovery feature in order to discover the pods in the cluster from which it gets metrics.
-In order to have this feature working, it is necessary for the service account used for running the Prometheus service pod to have access to the API server to get the pod list. By default the service account `prometheus-server` is used.
+* A `ClusterRole` that grants permissions to read Prometheus health endpoints of the Kubernetes system, including cAdvisor and kubelet for container metrics.  The Prometheus server configuration uses the Kubernetes service discovery feature in order to discover the pods in the cluster from which it gets metrics.  In order to have this feature working, it is necessary for the service account used for running the Prometheus service pod to have access to the API server to get the pod list.
+* A `ServiceAccount` for the Prometheus pods to run under.
+* A `ClusterRoleBinding` which binds the aforementioned `ClusterRole` to the `ServiceAccount`.
+* A `Deployment` to manage the actual Prometheus server pod.
+* A `ConfigMap` to manage the configuration of Prometheus Server.
+* A `Service` to provide an easy to reference hostname for other services to connect to Prometheus server (such as Grafana).
 
-[source,shell]
-export NAMESPACE=[namespace]
-oc login -u system:admin
-oc create sa prometheus-server
-oc adm policy add-cluster-role-to-user cluster-reader system:serviceaccount:${NAMESPACE}:prometheus-server
-oc login -u developer
-
-where `[namespace]` is the namespace/project where the Apache Kafka cluster was deployed.
-
-Finally, create the Prometheus service by running
-
-[source,shell]
-oc apply -f https://raw.githubusercontent.com/strimzi/strimzi-kafka-operator/master/metrics/examples/prometheus/kubernetes.yaml
+==== Deploying on {OpenShiftName}
 
-==== Grafana
+To deploy all these resources you can run the following.  Note that this file creates a `ClusterRoleBinding` in the `myproject` namespace.  If you're not using this namespace then download the resource file locally and update it.
```

Comment:
If this file will be included in what the user downloads why are we telling them to apply it from a URL (thus assuming the user has internet access)?

Reply:
TBH, I do not think we should provide the Grafana and Prometheus YAMLs in releases. From my perspective these to be used for testing etc. Users should they own production grade setup. Or are these deployments _production ready_ @seglo?

But we should maybe do it for the JSONs with Grafana dashboards.

Reply:
@tombentley @scholzj At the moment the Strimzi download doesn't bundle the `./metrics/examples` so the docs have the user downloading resource artifacts directly from the internet.

I think it might make sense to include it in the release bundle, again as an example.  Since we serve other example resources in that bundle I think it would fit in.  Then we could reference the resources there instead of the internet, and apply the same namespace `sed` instructions that are done in the cluster provisioning sections of the docs.  Would you lke me to update the release management to include it?

This monitoring setup everything you need in production, but I would say that the dashboard gets you about 75% of the way.  The missing pieces IMO are setting up alerts and any custom tweaks required by the client to accommodate their particular infrastructure.  

Another dimension this setup doesn't satisfy is _application_ monitoring.  The dashboards are a good starting point for infrastructure monitoring of the cluster itself, but monitoring Kafka application attributes such as consumer lag is a major component of a complete streaming monitoring setup.  I have some ideas here which I'll follow up on in another thread.

So while this setup isn't everything you need in production, it's a good starting point.  There's a lot of components and configuration required to make metrics available to the dashboards.  I think it's important to include working Prometheus JMX Exporter configurations and a Prometheus  server configuration so that users can get up and running out of the box.

Reply:
The reason we don't include metrics currently is because it don't want users to mistakenly think we're able to support their prometheus/grafana deployment. We're not in a position to do that, so we don't distribute those things and merely link to the them in the docs. 

We could stick to that approach, but I'm not sure how helpful it really is to the end user. An alternative would be to rearrange the current `examples` directory to better reflect things. Right now it contains `install` resources as well as more exemplar custom resources. If we has `install` (containing the current `examples/install`), `examples` and `contrib` top level directories, I think that might better reflect the status of these different things.

Reply:
@seglo I do not think the dashboards are something what would not be production ready (although its obviously not everything). But for example the Prometheus deployment YAML is using `emptyDir` volume which is probably not the right way for your production Prometheus deployment. So IMHO we need to make everyone aware that these are only if you want to try the monitoring and that this is not in any way something you should take to production just like that.

Reply:
@scholzj That's true.  The opening paragraph for the Metrics appendix section mention that the metrics setup isn't "recommended configuration", but I'll update this to include some disclaimers about the things like the Prometheus config, etc.  The main parts we should highlight as required for the Grafana dashboards to work is the Prometheus service discovery and relabeling config and the JMX exporter rules.

@tombentley I think your proposed structure is a good idea, but I'll keep it as-is for this PR.  Do you agree?

Reply:
@seglo fine with me. I'm interested in @scholzj's opinion of that idea.

Reply:
Yeah, I think the new structure is good. But keeping it for separate PR makes sense.